shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -235,11 +235,6 @@ - (BOOL)isFirstResponder {
 
 - (void)dealloc {
   [_channel setMethodCallHandler:nil];
-  if (_textInputContext) {
-    [_textInputContext deactivate];
-    [_textInputContext discardMarkedText];
```

Comment:
Are you sure there's no situation where the IME will retain mark text after delloc here? My guess is that since the mark text is maintained as part of the NSTextInputClient (in our case the FlutterTextInputPlugin) that the mark text is cleared anyway since the deallocation will prune it away anyway, but since we don't have access to the implementation details of NSTextInputContext, I'm a little hesitant to remove the call to clear the mark text.

Perhaps just eliminate the `deactivate` call, but keep the `discardMarkedText` call here.

Reply:
There are a couple more activate/deactivate calls in this file that we may want to eliminate too.

Reply:
> There are a couple more activate/deactivate calls in this file that we may want to eliminate too.

Searching `discardMarkedText` with [SourceGraph](https://sourcegraph.com/search?q=context:global+discardMarkedText&patternType=literal) reveals some usages to deactivate a `NSTextInputContext` deactivation. Most typical use case looks like this:
```objc
[inputContext discardMarkedText];
[inputContext invalidateCharacterCoordinates];
[inputContext deactivate];
```
But none of those open-source projects do these steps inside `dealloc`.

After some more manual testings, I can conclude that this issue is caused by `discardMarkedText` rather than `deactivate` call.

It might be reasonable to remove the entire block on `dealloc`, but I'm wondering how I can actually test this case.